hooks/post-command

```diff
@@ -49,42 +54,60 @@ plugin_read_list_into_result() {
 }
 
 get_codecov_uploader() {
-  if [ "${OS}" = "alpine" ] || [ "${OS}" = "linux" ] || [ "${OS}" = "macos" ]; then
-    if [ "${OS}" = "alpine" ]; then
-      apk add gnupg
-    fi
-    curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import # One-time step
-    for file in codecov codecov.SHA256SUM codecov.SHA256SUM.sig
-    do
-      if [[ ! -e "${TMP_DIR}/${file}" ]]; then
-        echo "${TMP_DIR}/${file}"
-        echo "https://uploader.codecov.io/${CODECOV_VERSION}/${OS}/${file}"
-        curl -s --create-dirs -o "${TMP_DIR}/${file}" "https://uploader.codecov.io/${CODECOV_VERSION}/${OS}/${file}"
-      fi
-    done
-
-    (
-      cd "${TMP_DIR}"
-      gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
-      if [ "${OS}" = "macos" ]; then
-        shasum -a 256 -c codecov.SHA256SUM
-      else
-        sha256sum -c codecov.SHA256SUM
-      fi
-      chmod +x codecov
-    )
-    return
+  case "${OS}" in
+    alpine|linux|macos)
+      # We support these
+      ;;
+    *)
+      error "Your platform (${OS}) is not supported."
+      ;;
+  esac
+
+  if [[ "${OS}" = "alpine" ]]; then
+    apk add gnupg
   fi
 
-  # Support for other OS such as windows will have to be implemented if needed
-  echo "Your platform (${OS}) is not supported."
-  exit 1
+  # One-time step
+  curl \
+      -fSs \
+      --retry 5 \
+      https://keybase.io/codecovsecurity/pgp_keys.asc \
+      | gpg \
+          --no-default-keyring \
+          --keyring trustedkeys.gpg \
+          --import
+  local file
+  for file in codecov codecov.SHA256SUM codecov.SHA256SUM.sig
+  do
+    local local_path="${TMP_DIR}/${file}"
+    if [[ ! -e "${local_path}" ]]; then
+      >&2 echo "Local path will be: ${local_path}"
+      local remote_source="https://uploader.codecov.io/${CODECOV_VERSION}/${OS}/${file}"
+      >&2 echo "Source is: ${remote_source}"
+      curl \
+          -fSs \
+          --create-dirs \
+          -o "${local_path}" \
+          "${remote_source}"
+    fi
+  done
+
+  (
+    pushd "${TMP_DIR}" >/dev/null
+    gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
+    if [[ "${OS}" = "macos" ]]; then
+      shasum -a 256 -c codecov.SHA256SUM
+    else
+      sha256sum -c codecov.SHA256SUM
+    fi
+    chmod +x codecov
+    popd >/dev/null
+  )
 }
 
 main() {
   if [[ "${BUILDKITE_PLUGIN_CODECOV_SKIP_ON_FAIL:-false}" =~ ^(true|on|1)$ ]] && [[ "${BUILDKITE_COMMAND_EXIT_STATUS}" -ne 0 ]]; then
-    echo "Codecov upload is skipped because step failed with status ${BUILDKITE_COMMAND_EXIT_STATUS}"
-    exit 0
```

Comment:
Another problem here, you are changing the exit code from 0 to 1. Now we have one failing test:
```
✗ Post-command is skipped if command failed and skip_on_fail=true
   (from function `assert_success' in file /usr/local/lib/bats/bats-assert/src/assert.bash, line 114,
    in test file tests/command.bats, line 62)
     `assert_success' failed
   
   -- command failed --
   status : 1
   output : Codecov upload is skipped because step failed with status 123
   --
```

Reply:
fixed in 92d2a04cb146f7ce071bce361582efa887338ee2